### PR TITLE
Add `AddGroup` API for ground groups

### PR DESCRIPTION
**Pull request by rurounijones**
Implement the `AddGroup` API for ground groups.  This PR is WIP but putting it up early for early feedback since it is going to be a monster.

Notes:
* My plan is to make the public gRPC API as nice to use as possible and then do the conversion to the required lua table in lua (e.g. Use the `Position` message in gRPC and then fill in the `x` and `y` values in the table from that Position)
* Some of the values are strings in lua like `AltitudeType` but should really have been Enums. In order to make things nicer for  clients I am going to try and turn as much as possible into Enums in gRPC then will change them back into strings in the lua implementation. I still need to find the list of options for some of them.
* I am creating separate Group and associated Unit templates for each Group Category because they are all different in terms of what fields are required vs optional.
* I have not filled out things like `Task` yet, still working on it.

https://wiki.hoggitworld.com/view/DCS_func_addGroup has a lot of useful information and example templates.

Implements #49

### CODE REVIEW

File: protos/dcs/coalition/coalition.proto [hunk 1]
```diff
@@ -23,6 +23,100 @@ service CoalitionService {
...
+	dcs.common.GroupCategory category = 2;
+	oneof template {
+	  GroundGroupTemplate ground_template = 3;
+	  ShipGroupTemplate ship_template = 4;
+	  HelicopterGroupTemplate helicopter_template = 5;
+	  PlaneGroupTemplate plane_template = 6;
+	}
+
+	message GroundGroupTemplate {
+		optional uint32 group_id = 1;
```
Comment (rkusa) [inline protos/dcs/coalition/coalition.proto:42]: Do you know what happens if the ID is already taken?

File: protos/dcs/coalition/coalition.proto [hunk 2]
```diff
@@ -23,6 +23,100 @@ service CoalitionService {
...
+		optional bool hidden = 2;
+		optional bool late_activation = 3;
```
Comment (rkusa) [inline protos/dcs/coalition/coalition.proto:44]: We might be able to simplify booleans by making `false` the default as we don't need them to be `optional` then. I think those two are already `false` by default, so I think we can ditch the `optional`?

File: protos/dcs/coalition/coalition.proto [hunk 3]
```diff
@@ -23,6 +23,100 @@ service CoalitionService {
...
+		string name = 4;
+		dcs.common.Position position = 5;
+		repeated Point route = 6;
+		optional uint32 start_time = 7;
+		string task = 8;
+		bool task_selected = 9;
+		repeated Task tasks = 10;
+		bool uncontrollable = 11;
+		repeated GroundUnitTemplate units = 12;
+		optional bool visible = 13;
```
Comment (rkusa) [inline protos/dcs/coalition/coalition.proto:54]: We also have `hidden` above - do they have different effects?

**Review by rkusa — COMMENTED** [review]
I like it so far! 👍

File: protos/dcs/coalition/coalition.proto (hunk 1 above)
Comment (rurounijones) [reply protos/dcs/coalition/coalition.proto:42]: I *think* it replaces the existing object but not entirely sure.

File: protos/dcs/coalition/coalition.proto (hunk 3 above)
Comment (rurounijones) [reply protos/dcs/coalition/coalition.proto:54]: Of course they are different effects! doesn't the naming make that _perfectly clear?!_ 😁

File: protos/dcs/coalition/coalition.proto (hunk 2 above)
Comment (rurounijones) [reply protos/dcs/coalition/coalition.proto:44]: Good point.

**Comment by rurounijones** [comment]
As of this commit we have _extremely_ basic ground group spawning.

**Comment by rurounijones** [comment]
With the code in this commit we are able to spawn a static SAM site and it will engage enemy targets. For example, the following request (sent using grpcui.exe) will spawn a NASAM site on the Caucasus map capable of engaging the enemy.

Request
```json
{
  "country": "COUNTRY_UNITED_STATES_OF_AMERICA",
  "groupCategory": "GROUP_CATEGORY_GROUND",
  "groundTemplate": {
    "name": "SAM Site",
    "position": {
      "lat": 42.4,
      "lon": 42.4
    },
    "route": [],
    "task": "Ground Nothing",
    "tasks": [],
    "units": [
      {
        "name": "SAM Site Launcher 1",
        "type": "NASAMS_LN_C",
        "position": {
          "lat": 42.4,
          "lon": 42.4
        },
        "skill": "SKILL_HIGH"
      },
      {
        "name": "SAM Site Launcher 2",
        "type": "NASAMS_LN_C",
        "position": {
          "lat": 42.4,
          "lon": 42.401
        },
        "skill": "SKILL_HIGH"
      },
      {
        "name": "SAM Site Launcher 3",
        "type": "NASAMS_LN_C",
        "position": {
          "lat": 42.4,
          "lon": 42.399
        },
        "skill": "SKILL_HIGH"
      },
      {
        "name": "SAM Site Radar",
        "type": "NASAMS_Radar_MPQ64F1",
        "position": {
          "lat": 42.399,
          "lon": 42.4
        },
        "skill": "SKILL_HIGH"
      },
      {
        "name": "SAM Site Command-Post",
        "type": "NASAMS_Command_Post",
        "position": {
          "lat": 42.398,
          "lon": 42.4
        },
        "skill": "SKILL_HIGH"
      }
    ]
  }
}
```
Response
```json
{
  "group": {
    "id": 1000006,
    "name": "SAM Site",
    "coalition": "COALITION_BLUE",
    "category": "GROUP_CATEGORY_GROUND"
  }
}

File: lua/methods/coalitions.lua [hunk 4]
```diff
@@ -1,11 +1,122 @@
...
+}
+
+--local altitudeType = {
+--  [1] = "BARO",
+--  [2] = "RADIO",
+--  BARO = 1,
+--  RADIO = 2
+--}
+
+local createGroundUnitsTemplate = function(unitListTemplate)
```
Comment (rkusa) [inline lua/methods/coalitions.lua:31]: Nit :D (feel free to ignore)

```suggestion
local function createGroundUnitsTemplate(unitListTemplate)
```

**Review by rkusa — APPROVED** [review]
What an undertaking 👍

File: lua/methods/coalitions.lua (hunk 4 above)
Comment (rurounijones) [reply lua/methods/coalitions.lua:31]: Plenty of PRs to come so will do this as part of them.

**Pull request merged**